atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetter.java

```diff
@@ -13,122 +13,34 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.palantir.atlasdb.keyvalue.dbkvs;
 
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
-import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbKvs;
 import com.palantir.common.exception.TableMappingNotFoundException;
-import com.palantir.logsafe.SafeArg;
-import com.palantir.logsafe.UnsafeArg;
-import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
 import java.util.Set;
-import java.util.stream.Collectors;
-import org.apache.commons.lang3.StringUtils;
-
-public final class OracleTableNameGetter {
-    private final String tablePrefix;
-    private final String overflowTablePrefix;
-    private final OracleTableNameMapper oracleTableNameMapper;
-    private final OracleTableNameUnmapper oracleTableNameUnmapper;
-    private final boolean useTableMapping;
-
-    private OracleTableNameGetter(
-            OracleDdlConfig config, OracleTableNameMapper tableNameMapper, OracleTableNameUnmapper tableNameUnmapper) {
-        this.tablePrefix = config.tablePrefix();
-        this.overflowTablePrefix = config.overflowTablePrefix();
-        this.useTableMapping = config.useTableMapping();
-
-        this.oracleTableNameMapper = tableNameMapper;
-        this.oracleTableNameUnmapper = tableNameUnmapper;
-    }
-
-    public static OracleTableNameGetter createDefault(OracleDdlConfig config) {
-        return new OracleTableNameGetter(config, new OracleTableNameMapper(), new OracleTableNameUnmapper());
-    }
-
-    public static OracleTableNameGetter createForTests(
-            OracleDdlConfig config, OracleTableNameMapper tableNameMapper, OracleTableNameUnmapper tableNameUnmapper) {
-        return new OracleTableNameGetter(config, tableNameMapper, tableNameUnmapper);
-    }
-
-    public String generateShortTableName(ConnectionSupplier connectionSupplier, TableReference tableRef) {
-        if (useTableMapping) {
-            return oracleTableNameMapper.getShortPrefixedTableName(connectionSupplier, tablePrefix, tableRef);
-        }
-        return getPrefixedTableName(tableRef);
-    }
-
-    public String generateShortOverflowTableName(ConnectionSupplier connectionSupplier, TableReference tableRef) {
-        if (useTableMapping) {
-            return oracleTableNameMapper.getShortPrefixedTableName(connectionSupplier, overflowTablePrefix, tableRef);
-        }
-        return getPrefixedOverflowTableName(tableRef);
-    }
 
-    public String getInternalShortTableName(ConnectionSupplier connectionSupplier, TableReference tableRef)
-            throws TableMappingNotFoundException {
-        if (useTableMapping) {
-            return oracleTableNameUnmapper.getShortTableNameFromMappingTable(connectionSupplier, tablePrefix, tableRef);
-        }
-        return getPrefixedTableName(tableRef);
-    }
+public interface OracleTableNameGetter {
```

Comment:
Aforementioned alternatives
* Allow mocking final
* Undo making it final
* Make TableNameUnmapper public and try and mock through that.